enterprise/server/githubapp/githubapp.go

```diff
@@ -1342,3 +1344,142 @@ func (a *GitHubApp) CreateGithubRef(ctx context.Context, req *ghpb.CreateGithubR
 
 	return &ghpb.CreateGithubRefResponse{}, nil
 }
+
+func (a *GitHubApp) GetGithubPullRequest(ctx context.Context, req *ghpb.GetGithubPullRequestRequest) (*ghpb.GetGithubPullRequestResponse, error) {
+	client, err := a.getGithubClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+	incoming, outgoing, user, err := a.getIncomingAndOutgoingPRs(ctx, client)
+	if err != nil {
+		return nil, err
+	}
+	allIssues := append(outgoing.Issues, incoming.Issues...)
+	prs, err := a.populatePRMetadata(ctx, client, allIssues, user)
+	if err != nil {
+		return nil, err
+	}
+	resp := &ghpb.GetGithubPullRequestResponse{
+		Outgoing: []*ghpb.PullRequest{},
+		Incoming: []*ghpb.PullRequest{},
+	}
+	for _, i := range outgoing.Issues {
+		resp.Outgoing = append(resp.Outgoing, prs[i.GetNumber()])
+	}
+	for _, i := range incoming.Issues {
+		resp.Incoming = append(resp.Incoming, prs[i.GetNumber()])
+	}
+	return resp, nil
+}
+
+func (a *GitHubApp) getIncomingAndOutgoingPRs(ctx context.Context, client *github.Client) (*github.IssuesSearchResult, *github.IssuesSearchResult, *github.User, error) {
+	var incoming *github.IssuesSearchResult
+	var outgoing *github.IssuesSearchResult
+	var user *github.User
+	searchGroup, searchContext := errgroup.WithContext(ctx)
+	searchGroup.Go(func() error {
+		r, _, err := client.Search.Issues(searchContext, "is:open is:pr user-review-requested:@me archived:false draft:false", &github.SearchOptions{ListOptions: github.ListOptions{PerPage: 100}})
+		if err != nil {
+			return err
+		}
+		incoming = r
+		return nil
+	})
+	searchGroup.Go(func() error {
+		r, _, err := client.Search.Issues(searchContext, "is:open is:pr author:@me -review:none archived:false draft:false", &github.SearchOptions{ListOptions: github.ListOptions{PerPage: 100}})
+		if err != nil {
+			return err
+		}
+		outgoing = r
+		return nil
+	})
+	searchGroup.Go(func() error {
+		u, _, err := client.Users.Get(ctx, "")
+		if err != nil {
+			return err
+		}
+		user = u
+		return nil
+	})
+	err := searchGroup.Wait()
+	if err != nil {
+		return nil, nil, nil, err
+	}
```

Comment:
if you don't need the err you can scope it
```
	if err := searchGroup.Wait(); err != nil {
		return nil, nil, nil, err
	}
```

Reply:
Done